### ISSUE TITLE
Anchor: Update examples to use href instead of onClick

### ISSUE THREAD

**Issue by elephantcatdog** [iTwin/iTwinUI-react#807]
`Anchor` is not keyboard accessible if there is only `onClick` and no `href` or `tabIndex` and `onKeyDown`. Components that use or rely on `Anchor` but use or encourage the use of `onClick` only should be revised to use or show keyboard accessible methods of using `Anchor`.

Two examples of this are in `Table` and `Alert`. The anchors in the Basic Table example only have `onClick`, which prevents them from being tabbed to. This can be fixed by either adding `href='#'` or by adding `tabIndex={0} onKeyDown={(e) => { if (e.key == 'Enter') { onClickHandler(props); } }}` Also the anchors in the Alert examples send the parameter `clickableTextProps={{ onClick: action('Clicked more info!') }}`, which has the same problem. This can be fixed in a similar way to the Table.

Revising these examples will encourage the correct use of Anchors as anchors instead of buttons. Or, if users are using them as buttons, at least they will be keyboard accessible. One problem with using hrefs in stories is that even an empty ('#') href will redirect to a new page on Storybook, and won't show as actions. 

But at least the description and example for `Alert` and any other components with similar usage of `Anchor` should be revised to using `clickableTextProps` with an href.

![Image](https://user-images.githubusercontent.com/37936169/187533726-457c5faf-414f-4d15-824c-5512ad8d6137.png)
